eng/Publishing.props

```diff
@@ -0,0 +1,28 @@
+<Project>
+
+  <PropertyGroup>
+    <!-- Don't push rid agnostic nuget packages from other builds than win-x64 when not building inside the VMR. -->
+    <EnableDefaultPublishItems Condition="'$(DotNetBuildRepo)' != 'true' and
+                                          '$(TargetArchitecture)' != 'x64' and
+                                          '$(TargetArchitecture)' != ''">false</EnableDefaultPublishItems>
```

Comment:
nit: instead of hardcoding win-x64 maybe it'd be better to set a property in the repo .yml in the x64 job to enable publishing?

Reply:
I'm not sure if that indirection would provide any benefit.